pliers/extractors/text_encoding.py

```diff
@@ -0,0 +1,812 @@
+        Dimensionality = “50|100|200|300” [for glove]; 300 
+        [for word2vec and dep2vec] (default:50)
+
+        corpus=”42B,” [for glove; default is 42B]. 
+        Note, unlike glove, the other models do not come up with so many choices. See README
+        for particular choice of corpus and diemensionality 
+
+        content_only = True|False [default:Content_only = True]
+        
+        binary = True|False [vector in binary or not, applicable for averageembedding]
+        
+        stopWords= None| list [default: NLTK stop words]
+
+        unk_vector (numpy array or str): Default vector to use for texts not
+            found in the embedding file. If None is specified, uses a
+            vector with all zeros. If 'random' is specified, uses a vector with
+            random values between -1.0 and 1.0. Must have the same dimensions
+            as the embeddings.
+            
+        layer (str description): For models such as ELMo user can select
+            a specific layer for encoding. This variable gives the user
+            the option of selecting a particular layer. Default (i.e., 
+            not specifying any value) is based on the model's specificity.
+    '''
+    
+    def __init__(self,method="averageWordEmbedding",\
+                             embedding="glove",\
+                             dimensionality=300 ,\
+                             corpus="6B",\
+                             content_only=True,\
+                             binary=False,\
+                             stopWords=None,\
+                             unk_vector=None,\
+                             layer=None):
+        
+        verify_dependencies(['keyedvectors']) 
+        verify_dependencies(['doc2vecVectors']) 
+        
+        '''Check the instance type of all the parameters'''
+        
+        if not isinstance(method,str):
+            raise ValueError('Method should be string (default is ' + \
+            'AverageWordEmbedding) or select' + \
+            ' from the list provided in README.')
+         
+        '''checking instance type of the parameters is done'''
+                
+        self.method = method
+        if self.method.lower() == "averagewordembedding":
+            
+            self.semvector_object = AverageEmbeddingExtractor(embedding=embedding,\
+                             dimensionality=dimensionality,\
+                             corpus=corpus,\
+                             content_only=content_only,\
+                             binary = binary,\
+                             stopWords=stopWords)
+            
+            self.semvector_object._loadModel()
+
+
+        elif method.lower() == 'skipthought':
+            
+            self.semvector_object = SkipThoughtExtractor()
+            self.semvector_object._loadModel() 
+            
+        elif method.lower() == 'sif':
+            
+            self.semvector_object = SmoothInverseFrequencyExtractor(embedding=embedding,\
```

Comment:
See my above comment, but I think it makes more sense to have these other specific extractors inherit from this base class, rather than be instantiated and set as instance variables by this class. 

Ideally, this pattern maximizes re-use of parallel code, and highlights the differences between the extractors. 

Additionally, this means that each specific extractor type is available externally (i.e. in `__init__`), which is useful when making pliers extraction graphs, etc..